validator/src/cli.rs

```diff
@@ -863,8 +863,8 @@ pub fn test_app<'a>(version: &'a str, default_args: &'a DefaultTestArgs) -> App<
                 .value_name("HOST")
                 .takes_value(true)
                 .validator(solana_net_utils::is_host)
-                .default_value("0.0.0.0")
-                .help("IP address to bind the validator ports [default: 0.0.0.0]"),
+                .default_value("127.0.0.1")
```

Comment:
Why is this change needed? I think this one is a big change

Reply:
test-validator normally runs on localhost, it is a dev tool. So binding to all interfaces by default was a bit excessive for it. All of this was to ultimately simplify socket binding logic in contact_info.rs